### ISSUE TITLE
Vim: Output is not a Terminal

### ISSUE THREAD

**Issue by AndiKod** [pyronaur/zxb#4]
Hi,
Congrats and thanks for ZXB, it's a great tool. Yet I run into an issue, maybe because of ZX itself:

It seems like I'm experiencing the case described in that article:
https://lxadm.com/vim-warning-output-is-not-to-a-terminal/

> The Output is not to a terminal warning occurs when Vim detects that its output is being redirected to a file or another program instead of being displayed on the terminal. This can happen when using Vim in a pipeline, in a script, or when redirecting its output to a file.

Tested on Fedora, Debian and Mint/WSL.

Maybe something about how [zx/process-output](https://google.github.io/zx/process-output) work. You can't even fire nano from a ZX script, it returns "Error: Standard input is not a terminal". I literally discovered ZX yesterday, so I'm maybe missing something but it's not a big deal. Just annoying.

tl;dr

To replace the edit() command, I just threw a little Bash script:
```bash
#!/usr/bin/bash

echo "What script will we edit?"
read script
vim ~/path/to/scripts/$script.mjs
```

...is it possible to do this using ZX? I got it with the question() part, but how can you write the last line?

**Comment by pyronaur** [comment]
👋  I honestly have no idea at this point. I've been out of the loop for zx for some time now - I'm instead actively working on bunmagic using Bun instead of node. I've found Bun to be much more friendly to writing shell-like scripts, especially because the dollar-shell syntax is a native Bun function (and I like TypeScript in my scripts too).

Migration to Bun and Bunmagic from zx and zxb should be pretty seamless for the most part:

https://www.npmjs.com/package/bunmagic
https://github.com/bunmagic/bunmagic

**Comment by AndiKod** [comment]
It was out of curiosity anyway. Bun is great indeed, as is Deno with their [compiler](https://docs.deno.com/runtime/reference/cli/compiler/) for specific targets. Creating Windows or Mac binaries out of TS files sounds like magic too.

Thx for the links, didn't looked enough into that side of Bun, I'll check out.

**Comment by pyronaur** [comment]
I'm not the expert on vim, but I think it should be possible with Bun - it'll just probably need a bit of redirection IIRC: https://bun.sh/docs/runtime/shell#redirection

I'm not a vim user, but if I recall correctly, I had to redirect vim through `/dev/tty` - but it's all fuzzy now 😄 

Good luck!
